app/Services/AppointmentService.php

```diff
@@ -118,9 +118,33 @@ public function reject($data, $appointmentId)
 
     public function checkAvailabiltyToReject($appointment)
     {
-        $nowInMs = Carbon::now()->timestamp * 1000;
+        $nowInMs = Carbon::now()->timestamp;
         if ($appointment->status != AppointmentStatus::PENDING || $appointment->time <= $nowInMs) {
             abort(Response::HTTP_BAD_REQUEST,"already rejected or expired");            
         }
     }
+
+    public function getCurrent()
+    {
+        $appointment = $this->repository->getCurrent();
+        if ($appointment->count() == 1) {
+            $appointment = $appointment->first();
+            return $appointment;
+        } else {
+            abort(Response::HTTP_BAD_REQUEST,'not found current appointment to checkin');
+        }
+    }
+    
+    public function checkCurrent($id)
+    {
+        $appointment = $this->show($id);
```

Comment:
You already queried for this appointment in the controller for the policy. No need to duplicate queries and just pass it to here